framework/src/constraints/AutomaticMortarGeneration.C

```diff
@@ -1727,21 +1775,33 @@ AutomaticMortarGeneration::projectSecondaryNodesSinglePair(
                  ++n)
               x2 +=
                   Moose::fe_lagrange_1D_shape(order, n, xi2_dn) * primary_elem_candidate->point(n);
-            auto u = x2 - (*secondary_node);
-            auto F = u(0) * nodal_normal(1) - u(1) * nodal_normal(0);
+            const auto u = x2 - (*secondary_node);
+            const auto F = u(0) * nodal_normal(1) - u(1) * nodal_normal(0);
 
             if (std::abs(F) < _newton_tolerance)
               break;
 
-            Real dxi2 = -F.value() / F.derivatives();
+            if (F.derivatives())
+            {
+              Real dxi2 = -F.value() / F.derivatives();
 
-            xi2_dn += dxi2;
+              xi2_dn += dxi2;
+            }
+            else
+              // It's possible that the nodal normal is completely orthogonal to the primary
+              // surface, in which case the derivative is 0. We know in this case that the
+              // projection should be a failure
```

Comment:
I don't understand this comment. Wouldn't that be a fairly common case in parallel plate contact?

Reply:
What's the implication that this case can happen commonly? I don't fully understand your comment.

Reply:
@dschwen I changed "orthogonal to the primary surface" to "orthogonal to the primary surface normal". Hopefully that clarifies things?

Reply:
and for good measure "nodal normal" to "secondary surface nodal normal"